doc/decisions/map_preloading.md

```diff
@@ -0,0 +1,51 @@
+# Map Preloading
+
+## Problem
+
+The application's main component is the map, which consists of multiple layers and even more objects. However, not all layers are visible simultaneously, nor are they required during the initial loading of the application. Loading all the data at once during the initial load may result in slower performance, therefore a strategy must be developed to preload data on the map to optimize its performance.
+
+## Constraints
+
+1.
+
+## Assumptions
+
+1.
+
+## Solutions
+
+### Full fetch
+
+Load all the data from the backend on the initial load. This will result in a slower initial load, but the map will be ready to use immediately.
```

Comment:
```suggestion
Load all the data from the backend on the initial load.
This will result in a slower initial load and probably very high memory consumption, but the map will be ready to use immediately.
```